File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        if not matrix:
            return
    
        rows = set()
        cols = set()
        
        # Find the rows and columns containing zeros
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if matrix[i][j] == 0:
                    rows.add(i)
                    cols.add(j)
        
        # Set entire rows to zero
        for row in rows:
            matrix[row] = [0] * len(matrix[row])
        
        # Set entire columns to zero
        for col in cols:
            for i in range(len(matrix)):
                matrix[i][col] = 0

File: 73-set-matrix-zeroes/set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    if (!matrix) return;

    let rows = new Set();
    let columns = new Set();

    for (let row = 0; row < matrix.length; row++) {
        for (let column = 0; column < matrix[row].length; column++) {
            if (matrix[row][column] === 0) {
                rows.add(row);
                columns.add(column);
            }
        }
    }

    for (let row = 0; row < matrix.length; row++) {
        for (let column = 0; column < matrix[row].length; column++) {
            if (rows.has(row) && columns.has(column)) {
                matrix[row][column] = 0;
            }
            if (rows.has(row)) {
                matrix[row][column] = 0;
            }
            if (columns.has(column)) {
                matrix[row][column] = 0;
            }
        }
    }
};